File: apps/api/src/utils/response.ts
import { Response } from 'express';

export function sendSuccess<T>(
  res: Response,
  data: T,
  message?: string,
  statusCode = 200,
  pagination?: { page: number; limit: number; total: number; totalPages: number },
) {
  res.status(statusCode).json({
    success: true,
    data,
    message,
    ...(pagination && { pagination }),
  });
}

export function sendError(res: Response, message: string, statusCode = 400) {
  res.status(statusCode).json({ success: false, message });
}

File: apps/api-go/internal/response/response.go
package response

import (
	"encoding/json"
	"net/http"
)

func Success(w http.ResponseWriter, status int, data any, message string) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	w.WriteHeader(status)
	_ = json.NewEncoder(w).Encode(map[string]any{"success": true, "data": data, "message": message})
}
func Paginated(w http.ResponseWriter, data any, page, limit, total int) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	w.WriteHeader(http.StatusOK)
	totalPages := 0
	if total > 0 {
		totalPages = (total + limit - 1) / limit
	}
	_ = json.NewEncoder(w).Encode(map[string]any{
		"success": true,
		"data":    data,
		"pagination": map[string]int{
			"page": page, "limit": limit, "total": total, "totalPages": totalPages,
		},
	})
}
func Error(w http.ResponseWriter, status int, message string) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	w.WriteHeader(status)
	_ = json.NewEncoder(w).Encode(map[string]any{"success": false, "message": message})
}
